Handle the case where `waitpid` returns `errno` `EINTR`.

Ivorforce opened the pull request:
Fixes #105097 

This case indicates that a debugger is attached, and `waitpid` should be called again.
Fix in reference to [this](https://stackoverflow.com/questions/21554195/why-waitpid-return-1-when-run-in-debugger) stackoverflow answer.
I'm not 100% sure this is the best way to handle it, but it seems to at least succeed in the debugger.

bruvzg reviewed (approved):
Tested on macOS, seems to be working.

`get_process_exit_code` and `is_process_running` probably should check for `-1`/`EINTR` as well and assume process is still running.

Ivorforce committed:
Handle the case where `waitpid` returns `errno` `EINTR`.
This case indicates that a debugger is attached, and `waitpid` should be called again.
Log errors when threads exit with `errno`.

Ivorforce commented:
@bruvzg This required a bit more refactoring, but ultimately i agree. Otherwise, we'll probably stumble over the same issue again some other time.

Ivorforce commented on drivers/unix/os_unix.cpp:
```diff
@@ -805,6 +805,55 @@ Dictionary OS_Unix::execute_with_pipe(const String &p_path, const List<String> &
+		return true;
+	}
+
+	ERR_FAIL_COND_V_MSG(result == -1, false, vformat("Thread {} exited with errno: {}", p_pid, errno));
```
Looks like I'm still sleeping 😄

bruvzg reviewed (approved):
Tested on macOS (with `lldb`) and Linux (with `lldb` and `gdb`) all methods seems to work as expected.

Repiteo commented:
Thanks!
